src/python/grpcio/grpc/_importer.py

```diff
@@ -0,0 +1,190 @@
+
+        # NOTE(rbellevi): module_repr is an abstract method in Python 3.3 only,b
+        #   but is still picked up by the linter.
+        class ProtoLoader(importlib.abc.Loader):  # pylint: disable=abstract-method
+
+            def __init__(self, module_name, protobuf_path):
+                self._module_name = module_name
+                self._protobuf_path = protobuf_path
+
+            def create_module(self, spec):
+                return None
+
+            def exec_module(self, module):
+                """Instantiate a module identical to the generated version.
+                """
+                # NOTE(rbellevi): include_paths are propagated via sys.path.
+                proto_module = protos(self._protobuf_path)
+                file_descriptor = getattr(proto_module,
+                                          _service_reflection.DESCRIPTOR_KEY)
+                for service_descriptor in file_descriptor.services_by_name.values(
+                ):
+                    _service_reflection.add_service_to_module(
+                        module, service_descriptor)
+
+        class ProtoFinder(importlib.abc.MetaPathFinder):
+
+            def find_spec(self, fullname, path, target=None):  # pylint: disable=no-self-use
+                del path
+                del target
+                filepath = _module_name_to_proto_file(fullname)
+                for search_path in sys.path:
+                    try:
+                        prospective_path = os.path.join(search_path, filepath)
+                        os.stat(prospective_path)
+                    except (FileNotFoundError, NotADirectoryError):
+                        continue
+                    else:
+                        return importlib.machinery.ModuleSpec(
+                            fullname, ProtoLoader(fullname, filepath))
+
+        def services(protobuf_path, *, include_paths=None):
+            """Loads gRPC service classes and functions from a .proto file.
+
+            THIS IS AN EXPERIMENTAL API.
+
+            Returns a module object equivalent to a "_pb2_grpc.py" file.
+
+            Together with the google.protobuf.protos function, this enables
+            deployment of services without a code generation step beforehand.
+
+            Args:
+              protobuf_path: A string representing the path to the desired
+                ".proto" file.
+              include_paths: A sequence of strings which should be searched for
+                the ".proto" file. By default, the entries on sys.path are
+                searched.
+
+            Returns:
+              A module object containing servicers, stubs, and associated
+              functions for the requested ".proto" file.
+            """
+            with _augmented_syspath(include_paths):
+                module_name = _proto_file_to_module_name(protobuf_path)
+                module = importlib.import_module(module_name)
+                return module
```

Comment:
Python searches modules in the order of `sys.path`. Is it consistent with ProtoBuf's behavior when there are multiple same name proto files?

Reply:
Yes. Protobuf's `SourceTreeDatabase` also searches its path mappings in order.